### ISSUE TITLE
KeePassXC doesn't properly switch between light and dark appearance on macOS

### ISSUE THREAD

**Issue by le-jawa** [keepassxreboot/keepassxc#7615]
## Overview
I've experienced a problem with KeePassXC on Macs (currently macOS 12) with the appearance set to change automatically.  If I run KeePassXC during the day (light appearance) and keep it running until after the appearance switches to dark, then the next morning (light again) KeePassXC will display with a mix of light and dark elements.

## Steps to Reproduce
1. Turn on automatic appearance switching on your Mac (macOS 10.15 or higher required)
2. Run KeePassXC during the day (light appearance) and leave it running until after the appearance switches to dark.
3. Shut down the Mac.
4. The next morning (light appearance) boot the Mac and open KeePassXC.  The interface will be a mix of light and dark elements.

## Expected Behavior
KeePassXC will switch between light and dark appearances flawlessly.

## Actual Behavior
The interface is a mix of light and dark elements when going to light appearance after KeePassXC has been running during the day and into the night, and then again the next morning.

## Context
This has happened since automatic appearance switching support was added in KeePassXC 2.6.4.
I've been able to work around the issue by manually switching my Mac appearance to "Light", then back to "Auto".

KeePassXC - VERSION 2.7.0
Revision: d7a9ef4

Operating System: macOS

![Screen Shot 2022-03-23 at 09 07 31](https://user-images.githubusercontent.com/19916394/159714117-86bd4e72-6984-4b93-8c1b-fed70f08af13.png)
![Screen Shot 2022-03-23 at 09 07 57](https://user-images.githubusercontent.com/19916394/159714146-a60a5643-4450-4e18-b9ed-ddef95dcaedf.png)

**Comment by droidmonkey** [comment]
Cannot reproduce this. 

![2022-03-23_10-45-51](https://user-images.githubusercontent.com/2809491/159726416-49982a6b-d678-43dd-9251-dbb831866402.gif)

This is both 2.6.6 and 2.7.0.

**Comment by le-jawa** [comment]
My work-around for the problem is to use General Settings applet; that "fixes" it, not causes it.  The only time I see this is when I do as reported above.  I realize that's a pretty long cycle for bug testing/reproduction, but that is when/how I encounter it.

I'll do some additional testing to see if I can find other avenues for reproduction, and will test on an additional Mac.  If I can find something that either shortens or clarifies the steps to reproduce, I'll post it here.

**Comment by varjolintu** [comment]
The `isDarkTheme()` check needs an extra check for the new Auto value. It should be a small fix. Testing it soon.

**Comment by phoerious** [comment]
I have used KeePassXC on macOS with auto theme switching for a long time and it has always worked perfectly.

**Comment by varjolintu** [comment]
@phoerious Including steps 3 and 4 where you shutdown you computer when the theme is dark and then boot back when it's already turned light?

**Comment by phoerious** [comment]
I haven't tried that, but it should send the same signals. Although I would consider it a real edge case. If you really shut down your Mac, it should restart KeePassXC on boot. Are you just putting your Mac in hibernate mode?

**Comment by varjolintu** [comment]
Here's an example in Swift how it's a bit [different](https://medium.com/@ruiaureliano/check-light-dark-appearance-for-macos-mojave-catalina-fb2343af875f) from the old Mojave implementation. The example doesn't check for the Auto value though.
https://github.com/ruiaureliano/macOS-Appearance/blob/master/Appearance/Source/AppDelegate.swift

**Comment by le-jawa** [comment]
Reproduced on my personal laptop last night and this morning.  (The other is my work laptop.)  Interesting, the screen shots didn't capture the "about" boxes.   That's 2.6.6. and 2.7.0.  

Since this is reproducible and it does appear to be a bug, can we re-label this a bug and remove the "not reproducible" label?

<img width="2048" alt="Screen Shot 2022-03-25 at 13 04 23" src="https://user-images.githubusercontent.com/19916394/160168539-07cc312e-1826-439d-879f-2ab4c30dd891.png">
<img width="2048" alt="Screen Shot 2022-03-25 at 12 36 43" src="https://user-images.githubusercontent.com/19916394/160168580-952d42ba-d9a8-47e5-96fc-ab138b398c1d.png">

**Comment by droidmonkey** [comment]
How did you reproduce it? The wait 12 hours while asleep method?

**Comment by le-jawa** [comment]
That depends on who's doing the sleeping.  ;-) 

I did just what I noted in the original report.  I first saw this on my work laptop, and the steps to recreate this are fairly normal routine for me.

1. I had appearance auto-switching on (it always is).
2.  I started my workday during daylight hours, and the light appearance was, of course, on.  I loaded KeePassXC, along with my other work apps and got to work.  In order to re-test this, I also started my personal laptop during the day, and opened KeePassXC.  I left my personal laptop (MacBook Pro) running all day.
3. That day happened to be a long one, so my laptop was still running, and KeePassXC open, when it got late, and my MacBook Air (my work laptop) switched from light to dark appearance.  KeePassXC successfully switched with it.  The same happened on my MacBook Pro.
4. After I was done for the evening, I let my Air go to sleep, and shutdown my Pro.
5. The next morning, I woke up my Air and KeePassXC had successfully switched back to the light appearance  (the behavior you're seeing).  When I booted my Pro, on the other hand, and started KeePassXC (both 2.6.6 and 2.7.0) it started with the mashup of the light and dark appearance you see in the screen shots above.  To get you the screenshots from my Air (the ones from the original report) I had shutdown my Air at night and restarted it the next morning.

Bottom line, you have to shut it down after the everything switches to the dark appearance, not put it to sleep.

This is common routine for me, so I don't "consider it a real edge case".  I have a hard time believing that Mac owners who are also KeePassXC users don't go through a similar routine and encounter the same issue.

**Comment by michaelk83** [comment]
@le-jawa can you reduce the test cycle by adjusting the transition schedule? So it would take 1-2 minutes or so?

**Comment by andkopp** [comment]
I have the same bug running macOS Monterey (12.4) and KeePassXC 2.7.1. Auto-switching is on. Normally my Macbook runs for weeks without shutting down. And like @le-jawa the first time I experienced this was when I had shutdown my Macbook Pro 16" in the evening (dark mode) and started it again in the morning - at the Apple Store for diagnostics. I didn't login there. First login was in the evening (dark mode again) and then I also got the mixed representation.

**Comment by opa334** [comment]
The issue is that there is a macOS bug where the `AppleInterfaceStyle` key in `.GlobalPreferences.plist` doesn't change when the system is shut down while the "automatic transition" happens, I have fixed this issue by not relying on that key for dark mode detection and will submit a PR shortly.

**Comment by pablosavigne** [comment]
Same issue MacOS 13.1 M1, just switch appearance to auto in Mac, and shutdown by night, and the next day will load in mixed dark. I'm changing this manually, since I installed the app!

**Comment by j-lakeman** [comment]
Can reproduce on macOS 11.6.8 Big Sur with KeepassXC 2.7.4 from Homebrew.
